src/dvclive/utils.py

```diff
@@ -125,3 +127,19 @@ def inside_notebook() -> bool:
 
         return IPython.__version__ >= "6.0.0"
     return False
+
+
+def get_unmatched_name_from_path(input_path, existing_names):
```

Comment:
Do we need it to be "unmatched"? For example, if I'm doing my own checkpointing, I might repeatedly call `log_artifact` with the same path and I'd expect it to overwrite each time.

Reply:
I think this was the reason for not using `name` as the key  (answer in the above comment 😅) .

What you describe works as expected in the current P.R. because we use `path` as the key and then check:

```
if path not in self._models:
```

To decide whether a `name` needs to be generated or not

Reply:
To clarify, we could achieve a similar thing using `name`, but the code felt worse to me.
Perhaps I am biased because I still feel that `path` made more sense as primary key in `artifacts` 😅

Reply:
Yeah, it feels to me like the code would be simpler if we use the name key and also don't care about unmatched names. It could infer a name if needed, then add whatever metadata is provided to `_models` as is, and write out `_models` as is to the artifacts section of `dvc.yaml`.

Also not really a blocker, but maybe worth a look since I wonder if the current implementation is biased too much by all the previous discussions and we will struggle to remember why it looks this way when we need to touch it in the future.

Reply:
> also don't care about unmatched names

Do you mean that we should not worry about:

```py
with Live() as live:
    live.log_artifact("classification/model.pth", type="model")
    live.log_artifact("detection/model.pth", type="model")
```

Reply:
Updated the P.R. to use `name` as primary key

Reply:
> Updated the P.R. to use `name` as primary key

TBH, it feels to me that we are moving the code from one place to another, not really simplifying

Reply:
Thanks for trying it this way @daavoo. Left some comments about where I think we can simplify.